worlds/dlcquest/Items.py

```diff
@@ -94,40 +97,65 @@ def create_trap_items(world, World_Options: Options.DLCQuestOptions, trap_needed
     return traps
 
 
-def create_items(world, World_Options: Options.DLCQuestOptions, locations_count: int, random: Random):
+def create_items(world, world_options: Options.DLCQuestOptions, locations_count: int, random: Random):
     created_items = []
-    if World_Options.campaign == Options.Campaign.option_basic or World_Options.campaign == Options.Campaign.option_both:
-        for item in items_by_group[Group.DLCQuest]:
-            if item.has_any_group(Group.DLC):
-                created_items.append(world.create_item(item))
-            if item.has_any_group(Group.Item) and World_Options.item_shuffle == Options.ItemShuffle.option_shuffled:
+    if world_options.campaign == Options.Campaign.option_basic or world_options.campaign == Options.Campaign.option_both:
+        create_items_basic(world_options, created_items, world)
+
+    if (world_options.campaign == Options.Campaign.option_live_freemium_or_die or
+            world_options.campaign == Options.Campaign.option_both):
+        create_items_lfod(world_options, created_items, world)
+
+    trap_items = create_trap_items(world, world_options, locations_count - len(created_items), random)
+    created_items += trap_items
+
+    return created_items
+
+
+def create_items_lfod(world_options, created_items, world):
+    for item in items_by_group[Group.Freemium]:
+        if item.has_any_group(Group.DLC):
+            created_items.append(world.create_item(item))
+        if item.has_any_group(Group.Item) and world_options.item_shuffle == Options.ItemShuffle.option_shuffled:
```

Comment:
(Can't believe I'm saying this, feel free to ignore it, it's painful to even think)
You can improve performance by doing the check for item shuffle outside of the loop
ughh I feel dirty now.